src/Cli/dotnet/commands/dotnet-workload/search/WorkloadSearchVersionsCommand.cs

```diff
@@ -123,5 +125,47 @@ public override int Execute()
 
             return 0;
         }
+
+        private List<string> GetVersions(int numberOfWorkloadSetsToTake)
+        {
+            var featureBand = new SdkFeatureBand(_sdkVersion);
+            var packageId = _installer.GetManifestPackageId(new ManifestId("Microsoft.NET.Workloads"), featureBand);
+
+            try
+            {
+                return PackageDownloader.GetLatestPackageVersions(packageId, numberOfWorkloadSetsToTake, packageSourceLocation: null, includePreview: _includePreviews)
+                    .GetAwaiter().GetResult()
+                    .Select(version => WorkloadSetVersion.FromWorkloadSetPackageVersion(featureBand, version.ToString()))
+                    .ToList();
+            }
+            catch (NuGetPackageNotFoundException)
+            {
+                Microsoft.DotNet.Cli.Utils.Reporter.Error.WriteLine(string.Format(LocalizableStrings.NoWorkloadVersionsFound, featureBand));
```

Comment:
I would probably not output a message from this method if no packages were found, and leave it to callers to handle that.  That's the behavior I would be more likely to expect when calling a method to get the versions, and it would let us have more specific messages for each case (ie "No workload sets were found for the current feature band" vs "No workload sets were found with the specified workload version(s)").